modules/example-browser/src/caching_materials_manager.ts

```diff
@@ -0,0 +1,194 @@
+   * that works with your application.
+   */
+  const { accessKeyId, secretAccessKey, sessionToken } = credentials
+
+  /* getClient takes a KMS client constructor
+   * and optional configuration values.
+   * The credentials can be injected here,
+   * because browser do not have a standard credential discover process the way Node.js does.
+   */
+  const clientProvider = getClient(KMS, {
+    credentials: {
+      accessKeyId,
+      secretAccessKey,
+      sessionToken
+    }
+  })
+
+  /* The KMS keyring must be configured with the desired CMKs */
+  const keyring = new KmsKeyringBrowser({ clientProvider, generatorKeyId, keyIds })
+
+  /* Create a cache to hold the material.
+   * In this case we use the local cache provided by the Encryption SDK.
+   * The number is the maximum number of entries that will be cached.
+   * Both encrypt and decrypt requests count independently towards this total.
+   * Elements will be actively removed from the cache.
+   * The default frequency is to check one item every minute.
+   * This can be configure by passing a `proactiveFrequency`
+   * as the second paramter to however often you want to check in milliseconds.
+   */
+  const cache = getLocalCryptographicMaterialsCache(100)
+
+  /* The partition name lets multiple caching CMMs share the same local cryptographic cache.
+   * If you want these CMMs to all cache the same items,
+   * make the partition name the same.
+   * If no partition is supplied a random one will be generated.
+   * This is so that sharing elements in the cache MUST be an intentional operation.
+   */
+  const partition = 'local partition name'
+
+  /* maxAge is the time in milliseconds that an entry will be cached.
+   * Elements are actively removed from the cache.
+   */
+  const maxAge = 1000 * 60
+
+  /* The maximum amount of bytes that will be encrypted under a single data key.
+   * This value is optional, but you should be configured to a lower value.
+   */
+  const maxBytesEncrypted = 100
+
+  /* The maximum number of messages that will be encrypted under a single data key.
+   * This value is optional, but you should be configured to a lower value.
+   */
+  const maxMessagesEncrypted = 100
+
+  const cmm = new WebCryptoCachingMaterialsManager({
+    backingMaterials: keyring,
+    cache,
+    partition,
+    maxAge,
+    maxBytesEncrypted,
+    maxMessagesEncrypted
+  })
+
+  /* Encryption context is a *very* powerful tool for controlling and managing access.
+   * It is ***not*** secret!
+   * Encrypted data is opaque.
+   * You can use an encryption context to assert things about the encrypted data.
+   * Just because you can decrypt something does not mean it is what you expect.
+   * For example, if you are are only expecting data from 'us-west-2',
+   * the origin can identify a malicious actor.
+   * See: https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#encryption-context
```

Comment:
Better:  The following doc describes the use of the encryption context in data key caching.
https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/data-caching-details.html#caching-encryption-context

Reply:
That link is not about cacheing, it is about encryption context.  i.e. why you should include/use it.